### ISSUE TITLE
Error while using makedirs from PyFilesystem

### ISSUE THREAD

**Issue by blasterspike** [tomgross/pycloud#10]
Hi,
I'm getting an error while trying to use makedirs from PyFilesystem with pycloud.
As per documentation from PyFilesystem
https://docs.pyfilesystem.org/en/latest/reference/base.html#fs.base.FS.makedirs

> Make a directory, **and any missing intermediate directories.**

Using this code

```
import fs
from pcloud import PyCloud
import urllib.parse

username = urllib.parse.quote_plus('my_username')
password = urllib.parse.quote_plus('my_password')

with fs.opener.open_fs('pcloud://{0}:{1}@/'.format(username, password)) as pcloud_fs:
    pcloud_fs.makedirs(path='/backup/second_level/third_level', recreate=True)
```

I'm getting the following error

```
Traceback (most recent call last):
  File "fs_makedirs.py", line 10, in <module>
    pcloud_fs.makedirs(path='/backup/second_level/third_level', recreate=True)
  File "/usr/local/lib/python3.6/site-packages/fs/base.py", line 1035, in makedirs
    dir_paths = tools.get_intermediate_dirs(self, path)
  File "/usr/local/lib/python3.6/site-packages/fs/tools.py", line 82, in get_intermediate_dirs
    resource = fs.getinfo(path)
  File "/usr/local/lib/python3.6/site-packages/pcloud/pcloudfs.py", line 118, in getinfo
    for item in folder_list['metadata']['contents']:
KeyError: 'metadata'
```

where:
/backup: it exists
/backup/second_level: it doesn't exist
/backup/third_level: it doesn't exist

and _getinfo_ tries to fetch _metadata_ of a directory that doesn't exist (parent_path = /backup/second_level).

Versions used in this test:
macOS 10.11.6
Python 3.6.5
fs==2.0.27
pcloud==1.0a4

Regards

Massimo

**Comment by tomgross** [comment]
Fixed with #11

**Comment by tomgross** [comment]
Closed with #11
